src/Internal.AspNetCore.Sdk/build/Git.targets

```diff
@@ -0,0 +1,63 @@
+﻿<Project>
+
+  <PropertyGroup>
+    <GenerateCommitHashAttribute Condition="'$(GenerateCommitHashAttribute)'==''">true</GenerateCommitHashAttribute>
+    <GeneratedCommitHashAttributeFile Condition="'$(GeneratedCommitHashAttributeFile)'==''">$(IntermediateOutputPath)$(AssemblyName).CommitHash$(DefaultLanguageSourceExtension)</GeneratedCommitHashAttributeFile>
+  </PropertyGroup>
+
+  <Target Name="ResolveCommitHash" Condition="'$(CommitHash)'==''">
+    <GetGitCommitInfo WorkingDirectory="$(MSBuildProjectDirectory)" ContinueOnError="WarnAndContinue">
+      <Output TaskParameter="CommitHash" PropertyName="CommitHash" />
+    </GetGitCommitInfo>
+  </Target>
+
+<!--
+********************************************************************************************
+Target: GenerateCommitHashAttribute
+Generates an assembly attribute with commit hash
+********************************************************************************************
+-->
+  <Target Name="GenerateCommitHashAttribute"
+          BeforeTargets="CoreCompile"
+          DependsOnTargets="ResolveCommitHash;PrepareForBuild;GenerateIntermediateCommitHash;CoreGenerateCommitHashAttribute"
+          Condition="'$(GenerateCommitHashAttribute)'=='true' and '$(DesignTimeBuild)'!='true'" >
+    <Warning Text="Property 'CommitHash' was not set"
+             Condition="'$(CommitHash)'==''" />
+  </Target>
+
+  <Target Name="GenerateIntermediateCommitHash" Condition="'$(CommitHash)' != ''">
+    <PropertyGroup>
+      <!-- shorten to help avoid max path length issues -->
+      <IntermediateCommitHash>$(IntermediateOutputPath)$(CommitHash.Substring(0, 10)).commit</IntermediateCommitHash>
```

Comment:
I don't see how this solves the problem. You would still need two files to make incremental build work.